### PR TITLE
Add basic UI for individual trip arrival time displays

### PR DESCRIPTION
**Pull request by dellisd**
Add framework for enabling grouping by routes, and other display items in the TripsAdapter (for the StopFragment).

# Screenshots

### CODE REVIEW

File: app/src/main/java/ca/llamabagel/transpo/data/TripsRepository.kt [hunk 1]
```diff
@@ -8,6 +8,8 @@ import ca.llamabagel.transpo.data.api.TripsService
 import ca.llamabagel.transpo.data.db.Stop
 import ca.llamabagel.transpo.data.db.TransitDatabase
 import ca.llamabagel.transpo.models.trips.ApiResponse
+import ca.llamabagel.transpo.ui.trips.TripAdapterItem
+import ca.llamabagel.transpo.ui.trips.TripUiModel
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/data/TripsRepository.kt]: unused imports

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/adapter/TripsAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    override fun getItemViewType(position: Int): Int = when (items[position]) {
+        is TripItem -> R.layout.trip
+    }
+
+    override fun getItemCount(): Int = items.size
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/ui/trips/adapter/TripsAdapter.kt]: Do we need a when statement for just 1 view type?

File: app/src/main/res/layout/trip.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="56dp">
+
+        <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:id="@+id/route_number"
+                app:layout_constraintStart_toStartOf="parent"
+                android:layout_marginStart="16dp"
```
Comment (GustavoSanMartin) [inline app/src/main/res/layout/trip.xml]: let's use dimensions

File: app/src/main/java/ca/llamabagel/transpo/data/TripsRepository.kt (hunk 1 above)
Comment (GustavoSanMartin) [reply app/src/main/java/ca/llamabagel/transpo/data/TripsRepository.kt]: so CI is not running lint checker?

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/adapter/TripsAdapter.kt (hunk 2 above)
Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/ui/trips/adapter/TripsAdapter.kt]: The adapter will use multiple ViewHolder types in the future, they just haven't been implemented yet. Specifically, there is a RouteViewHolder which will handle the display of arrivals when grouped by routes.

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsViewModel.kt [hunk 4]
```diff
@@ -21,8 +21,8 @@ class TripsViewModel @Inject constructor(private val tripsRepository: TripsRepos
     private val _stop = MutableLiveData<Stop?>()
     val stop: LiveData<Stop?> = _stop
 
-    private val _apiResponse = MutableLiveData<ApiResponse>()
-    val apiResponse: LiveData<ApiResponse> = _apiResponse
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsViewModel.kt]: unused import

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/adapter/TripsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+package ca.llamabagel.transpo.ui.trips.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import ca.llamabagel.transpo.R
+import ca.llamabagel.transpo.databinding.RouteBinding
+import ca.llamabagel.transpo.databinding.TripBinding
+import ca.llamabagel.transpo.models.trips.ApiResponse
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/ui/trips/adapter/TripsAdapter.kt]: unused import

**Pull request merged**